src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs

```diff
@@ -62,7 +62,7 @@ private void AnalyzeManagedAssemblyAndPdb(BinaryAnalyzerContext context)
             PEBinary target = context.PEBinary();
             Pdb di = target.Pdb;
 
-            if (target.PE.ManagedPdbSourceFileChecksumAlgorithm(di.FileType) != ChecksumAlgorithmType.Sha256)
+            if (target.PE.ManagedPdbSourceFileChecksumAlgorithm(di.FileType, di) != ChecksumAlgorithmType.Sha256)
```

Comment:
>ManagedPdbSourceFileChecksumAlgorithm [](http://example.com/codeflow?start=26&length=37)

can you verify if we have enough coverage for the ba2004 rule?

Reply:
yes, we do have 14 test files for this rule with combination of:
full_exe_sha1.exe
full_sha1.dll
portable_exe_sha1.exe
portable_sha1.dll
netcoreapp3.1
netstandard2.0
Native